src/test/java/io/managed/services/test/kafka/KafkaMgmtAPITest.java

```diff
@@ -218,6 +218,123 @@ public void testCreateTopics() {
+            assertThrows(ApiGenericException.class, () -> kafkaInstanceApi.createTopic(payloadTopicMax));
+        } else {
+            // create two topics, first will reach the max limit, creation of second will exceed it.
+            log.info("Trying to create topics with {} and 1 partitions, while total partition limit is {}, and instance already has {} partitions", maxPartitionLimit, maxPartitionLimit, currentPartitionCount);
+            // first topic
+            kafkaInstanceApi.createTopic(payloadTopicMax);
+            // second topic
+            var payloadAdditionalTopic = new NewTopicInput()
+                    .name("additional")
+                    .settings(new TopicSettings().numPartitions(1));
+            assertThrows(ApiGenericException.class, () -> kafkaInstanceApi.createTopic(payloadAdditionalTopic));
+        }
+
+        // cleanup
+        try {
+            kafkaInstanceApi.deleteTopic(topicName);
+        } catch (Throwable t) {
+            log.warn("clean {} topic error: ", topicName, t);
+        }
+
+        // there is a need to wait for some period of time before continue with other test as they still may need to create topics (i.e. extra partitions)
+        log.info("wait 15 seconds before making sure changes in partition count were propagated");
+        Thread.sleep(ofSeconds(15).toMillis());
+    }
+
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testTotalPartitionLimitByTopicConfiguration() {
+
+        final var topicFillPartitionCountName = "topic-fill-partition-count";
+        final var topicName = "topic-part-total-exceed-conf";
+
+        log.info("Apply topic with 1 partition");
+        KafkaInstanceApiUtils.applyTopic(kafkaInstanceApi, topicName);
+
+        log.info("Create topic with partitions so that there is still 1 more partition within limit");
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        final int currentPartitionCount = KafkaInstanceApiUtils.getPartitionCountTotal(kafkaInstanceApi);
+        var payloadTopicPartitionFiller = new NewTopicInput()
+                .name(topicFillPartitionCountName)
+                .settings(new TopicSettings().numPartitions(maxPartitionLimit - currentPartitionCount - 1));
+        kafkaInstanceApi.createTopic(payloadTopicPartitionFiller);
+
+        // there is a need to wait for some period of time before actually,
+        log.info("wait 15 seconds before making sure changes in partition count were propagated");
+        Thread.sleep(ofSeconds(15).toMillis());
+
+        log.info("Increase partition count on topic to reach the total limit");
+        KafkaInstanceApiUtils.updateTopicPartition(kafkaInstanceApi, topicName, 2);
```

Comment:
If my reading of this test is correct, then I'd expect the exception to be thrown at this point. Here's the flow as I see it:
1. L297 creates a topic with 1 partition
2. L305 creates a topic with "limit -1" partitions (so including the partition created earlier, the total number of partitions is at the limit now).
3. Here, L312 increases the original topic to 2 partitions (which would go over the limit). Since there's a sleep before this request, I think the partition budget should have been reduced to 0 already, and the authorizer would deny the request to increase.

Does that sound correct?

Reply:
1.) correct
2.) lets say we have only topic created in first step (line 297 and call it topic-1) with 1 partition, and limit of 1500 partitions.    (maxPartitionLimit - currentPartitionCount - 1) should translate to 1500 - 1 - 1. That is 1498, which is the number of partitions in newly created topic (lets call it topic-1498)
3.) now we have topic-1 with 1 partition, topic-1498 with 1498 partitions. Thus line 312 should actually reach the limit 1500 (by topic-1 holding now 2 partitions) but still be okay if I understand it correctly. 

Point 3 is just my understanding I did not manage to run it properly in stage, i.e., I only think that when I have 1499 taken and limit is 1500 I am still able to for example create topic with 1 partition.

Reply:
Ah yes, I think you're right, thanks :+1: